### PR TITLE
Intricate logic for points

### PR DESCRIPTION
**Pull request by Mafusn**
## Description

Logic for points when answering exercises.

## Changes

- Updated the studentmodel to show completionDate for exercises, sections and courses. 
- Implemented logic for answering exercises and give points based thereof

## Checklist

- [x] Code has been tested locally and passes all relevant tests.
- [x] Documentation has been updated to reflect the changes, if applicable.
- [x] Code follows the established coding style and guidelines of the project.
- [x] All new and existing tests related to the changes have passed.
- [x] Any necessary dependencies or new packages have been properly documented.
- [x] Pull request title and description are clear and descriptive.
- [x] Reviewers have been assigned to the pull request.
- [x] Any potential security implications have been considered and addressed.
- [x] Performance impact of the changes has been evaluated, if relevant.

### CODE REVIEW

File: routes/studentRoutes.js [hunk 1]
```diff
@@ -256,29 +260,60 @@ async function markExerciseAsCompleted(user, courseId, sectionId, exerciseId) {
...
-      const isExerciseAlreadyCompleted = user.completedCourses[completedCourseIndex].completedSections.some(completedSection =>
-        completedSection.sectionId.equals(sectionId) &&
-        completedSection.completedExercises.some(completedExercise => completedExercise.exerciseId.equals(exerciseId))
-      );
-
-      // Returns if the exercise is already completed
-      if (isExerciseAlreadyCompleted) {
-        throw errorCodes['E0801'];
-      }
+      let exerciseFound;
```
Comment (Jatewo) [inline routes/studentRoutes.js:269]: These functions should have a complete rework at some point - they are not the most readable 😄 

But let's add it to our technical debt 👍

**Review by Jatewo — APPROVED** [review]
### Nice work

Looking good, only a small comment on the readability on studentRoutes but no changes needed for now

File: routes/studentRoutes.js (hunk 1 above)
Comment (Mafusn) [reply routes/studentRoutes.js:269]: I agree

**Review by TheMOHD3886 — APPROVED** [review]
Looks good 👍

**Pull request merged**